third_party/2and3/typing_extensions.pyi

```diff
@@ -0,0 +1,28 @@
+import sys
+import typing
+from typing import ClassVar as ClassVar
+from typing import ContextManager as ContextManager
+from typing import Counter as Counter
+from typing import DefaultDict as DefaultDict
+from typing import Deque as Deque
+from typing import NewType as NewType
+from typing import overload as overload
+from typing import Text as Text
+from typing import TYPE_CHECKING as TYPE_CHECKING
+
+if sys.version_info >= (3, 3):
+    from typing import ChainMap as ChainMap
+
+if sys.version_info >= (3, 5):
+    from typing import AsyncIterable as AsyncIterable
+    from typing import AsyncIterator as AsyncIterator
+    from typing import AsyncContextManager as AsyncContextManager
+    from typing import Coroutine as Coroutine
+
+if sys.version_info >= (3, 6):
+    from typing import AsyncGenerator as AsyncGenerator
+
+# Return type that indicates a function does not return.
+# This type is equivalent to the None type, but the no-op Union is necessary to
+# distinguish the None type from the None value.
+NoReturn = typing.Union[None]
```

Comment:
Hm, any reason it isn't in `typing.pyi` yet? Maybe we can just fix that in this PR. If you'd prefer not to, I'll just merge this.